src/Routes/RosPage/RosPage.js

```diff
@@ -26,25 +51,92 @@ class RosPage extends React.Component {
         super(props);
 
         this.state = {
-            page: 1,
-            perPage: 10
+            perPage: 10,
+            orderBy: 'display_name',
+            orderDirection: SortByDirection.asc,
+            columns: [
+                { key: 'display_name', title: 'System name', renderFunc: systemName },
+                { key: 'display_performance_score.cpu_score', title: 'CPU score', renderFunc: scoreProgress },
+                { key: 'display_performance_score.memory_score', title: 'Memory score', renderFunc: scoreProgress },
+                { key: 'display_performance_score.io_score', title: 'I/O score', renderFunc: scoreProgress },
+                { key: 'recommendation_count', title: 'Recommendations',
+                    renderFunc: recommendations, props: { isStatic: true } },
+                { key: 'state', title: 'State', props: { isStatic: true } }
+            ]
         };
 
+        this.inventory = React.createRef();
+        this.chunkSize = 50;
+        this.fetchSystems = this.fetchSystems.bind(this);
     }
 
-    async componentDidMount() {
+    async fetchSystems(fetchParams) {
         await window.insights.chrome.auth.getUser();
-        this.props.fetchSystems({ page: this.state.page, perPage: this.state.perPage });
+        let params = {};
+        params.limit = fetchParams.perPage;
+        params.offset = (fetchParams.page - 1) * fetchParams.perPage;
+        params.order_by = fetchParams.orderBy || this.state.orderBy; /* eslint-disable-line camelcase */
+        params.order_how = fetchParams.orderHow || this.state.orderDirection; /* eslint-disable-line camelcase */
+
+        if (fetchParams.filters && fetchParams.filters.hostnameOrId) {
+            params.display_name =  fetchParams.filters.hostnameOrId; /* eslint-disable-line camelcase */
+        }
```

Comment:
This is rather expensive as it requires multiple assignments and memory calculation, much cleaner and faster approach is by assigning directly to params
```suggestion
        const params = {
          limit: fetchParams.perPage,
          offSet: (fetchParams.page - 1) * fetchParams.perPage,
          order_by: fetchParams.orderBy || this.state.orderBy, /* eslint-disable-line camelcase */
          order_how: fetchParams.orderHow || this.state.orderDirection, /* eslint-disable-line camelcase */
          ...fetchParams?.filters?.hostnameOrId && {
            display_name: fetchParams.filters.hostnameOrId; /* eslint-disable-line camelcase */
          }
        };
```